### PR TITLE
feat(guide-reader): add document outline navigation (jump-to-heading rail)

### LINKED ISSUE #1004: feat: External browser tab mode for custom guides
> Having content "right there" next to the rest of the Grafana UI is cool, albeit space is at a premium so sometimes it's just nicer to have it in a separate window (especially if I have a second monitor) 

We currently have a "go to website feature" for docs pages, which will open the origin docs in a new browser tab. Perhaps we have a method for doing this for private guides. This mode would be strickly readonly and no interaction.

### PR DESCRIPTION
**Pull request by Jayclifford345**
## Summary

Adds a GitHub-style document outline (jump-to-heading rail) to the two-tab guide reader: heading/section extraction with generated anchor ids, a sticky rail with click-to-jump, scroll-spy highlighting of the current section, and keyboard/a11y support.

https://github.com/user-attachments/assets/89ef663c-4d31-42b9-8c7f-240245ee2b31

## What to look at

- `src/utils/slug.ts` + `src/hooks/useDocumentOutline.ts` — GitHub-style slugify (diacritics stripped, `-1`/`-2` collision suffixer) and a hook that queries `h2, h3, h4, [data-interactive-section="true"]` in document order, assigning generated ids onto headings that don't already have one. When a plain heading is immediately followed by an interactive section sharing the same title text (the common `## Explore your data` + `<InteractiveSection title="Explore your data">` authoring pattern — see `src/bundled-interactives/first-dashboard/content.json`), the heading entry collapses into the section entry so the outline doesn't show the same title twice in a row.
- `src/components/guide-reader/GuideReaderOverlay.tsx` — restructures the backdrop's single content column into a `layoutRow` flex row (content column + rail), adds a `contentReady` flag flipped by `ContentRenderer`'s `onContentReady` callback (reset via the "adjust state during render when a prop changes" pattern rather than a `useEffect` + `setState`, since this repo's `react-hooks/set-state-in-effect` lint rule flags the latter), and a `backdropRef` used as the scroll-spy's `IntersectionObserver` root.
- `src/components/guide-reader/OutlineRail.tsx` — click-to-jump (`jumpToHeading`: container-scoped `querySelector` + smooth `scrollIntoView({ block: 'start' })` + the existing `fragment-highlight` pulse, mirroring `scrollToFragment` in `content-renderer.tsx` without touching that function, since the URL hash is reserved for controller pairing); `aria-current` + active styling on the current item; roving `ArrowUp`/`ArrowDown` focus between the list's native `<button>`s.
- `src/hooks/useActiveOutlineItem.ts` — one `IntersectionObserver` per outline render, `rootMargin` carving out the top 25% of the viewport as the "active" band; on every callback it recomputes from live `getBoundingClientRect()` and picks the *last* item whose top has crossed the band (ties break to the later item). `notifyJump(id)` sets `activeId` immediately and suppresses spy recomputes for 1s so a clicked item stays highlighted through its own smooth-scroll animation instead of flickering mid-scroll.
- The rail is `position: sticky` inside the backdrop's own scroll container and hides below `1280px` (v1: hidden, not collapsible).
- This entire surface is dark: `GuideReaderOverlay` only mounts when the `enableTwoTabController` admin setting is on (default off), so there's no user-visible change without that opt-in.

## Why

The two-tab reader is the only Pathfinder surface with enough horizontal room (a full browser tab, 1100px content column) for an outline rail; the narrow sidebar docs panel is explicitly out of scope. Building it here means it ships behind the existing admin toggle with no new gating code. The scroll-spy design (top-quarter band, last-crossed-wins, suppress-during-jump) matches GitHub's file outline behavior: the highlighted section tracks scroll position, and clicking doesn't fight the spy while the page is still animating to the target.

## How to verify

1. Unit/component tests cover the extraction rules, click/empty-state rendering (`scrollIntoView` mocked — jsdom doesn't implement it), and the scroll-spy band-crossing/tie-break/suppression logic (`IntersectionObserver` mocked — jsdom has none too).
2. Manual verification in a real Grafana instance (rail rendering, click-jump, scroll-spy, keyboard nav, narrow-viewport hide) was not performed in this session — flagging for reviewer or a follow-up manual pass. This surface's E2E coverage is already deferred elsewhere pending the `enableTwoTabController` flag flip.

### CODE REVIEW

File: src/locales/cs-CZ/grafana-pathfinder-app.json [hunk 1]
```diff
@@ -81,5 +81,8 @@
       "simplifiedView": "This is a simplified view. You can replace this with Grafana's full HelpModal component.",
       "toggleViewMode": "Toggle view mode"
     }
+  },
+  "guideReader": {
```
Comment (moxious) [inline src/locales/cs-CZ/grafana-pathfinder-app.json]: What's the point of so many repeated i18n entries for different languages you don't populate?  Seems like wasted extra lines/complexity for no benefit, falling back to nothing.

File: src/hooks/useDocumentOutline.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  for (const el of elements) {
+    const isSection = el.getAttribute('data-interactive-section') === 'true';
+    const text = (isSection ? el.querySelector(SECTION_TITLE_SELECTOR)?.textContent : el.textContent)?.trim() ?? '';
+    if (!text) {
+      continue;
+    }
+
+    let id = el.id;
+    if (!id) {
+      id = uniqueSlug(slugify(text), taken);
```
Comment (moxious) [inline src/hooks/useDocumentOutline.ts]: **(blocking)** `slugify` only strips Latin diacritics (NFKD), so any heading with no ASCII alphanumerics slugifies to `""`. A non-empty heading with an empty slug still gets pushed, yielding an empty `el.id`; `useActiveOutlineItem` then runs `querySelector('#' + CSS.escape(''))` → selector `'#'` → uncaught `SyntaxError`, which blanks the reader. `OutlineRail.jumpToHeading` throws the same way on click.

Suggested fix — guarantee a non-empty base slug:
```suggestion
      id = uniqueSlug(slugify(text) || 'section', taken);
```
Please add a regression test with a CJK/Cyrillic/emoji heading asserting non-empty ids and that the spy hook doesn't throw.

File: src/hooks/useActiveOutlineItem.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    if (tracked.length === 0) {
+      return;
+    }
+
+    const recompute = () => {
+      if (Date.now() < suppressedUntilRef.current) {
+        return;
+      }
+      const rootTop = root.getBoundingClientRect().top;
+      const bandHeight = root.getBoundingClientRect().height * TOP_BAND_FRACTION;
```
Comment (moxious) [inline src/hooks/useActiveOutlineItem.ts]: **(nit)** Two forced-layout reads on the same element per callback. Read the rect once:
```suggestion
      const rootRect = root.getBoundingClientRect();
      const rootTop = rootRect.top;
      const bandHeight = rootRect.height * TOP_BAND_FRACTION;
```

File: src/hooks/useActiveOutlineItem.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      return;
+    }
+
+    const tracked = items
+      .map((item) => ({ item, el: container.querySelector<HTMLElement>('#' + CSS.escape(item.id)) }))
+      .filter((entry): entry is { item: OutlineItem; el: HTMLElement } => entry.el !== null);
+
+    if (tracked.length === 0) {
+      return;
+    }
```
Comment (moxious) [inline src/hooks/useActiveOutlineItem.ts:35]: **(nit)** When `items` is non-empty but no ids resolve (`tracked.length === 0`), this returns without resetting `activeId`, unlike the `items.length === 0` guard above — a stale highlight could linger across a content swap. Suggest:
```suggestion
    if (tracked.length === 0) {
      setActiveId(null);
      return;
    }
```

**Review by moxious — CHANGES_REQUESTED** [review]
Nice feature — clean architecture, DOM work kept in effects, `CSS.escape` used throughout, and good test coverage (31/31 passing locally). One correctness issue to fix before merge, plus two cheap cleanups worth folding in.

**Blocking:** headings with no ASCII alphanumerics (CJK, Cyrillic, Greek, Arabic, Hebrew, emoji) slugify to an empty string, produce an empty DOM id, and then crash the scroll-spy with an uncaught `SyntaxError: '' is not a valid selector`. Reproduced locally: Japanese/Cyrillic/emoji headings → ids `["", "-1"]` → throw in the active-item effect. `OutlineRail.jumpToHeading` throws the same way on click. One-line fallback fixes it; details inline.

File: src/hooks/useDocumentOutline.ts (hunk 2 above)
Comment (Jayclifford345) [reply src/hooks/useDocumentOutline.ts]: Fixed in b605fd0d with exactly the suggested fallback (`uniqueSlug(slugify(text) || 'section', taken)`), plus the regression test you asked for: a CJK/Cyrillic/emoji-only heading case in `useDocumentOutline.test.ts` asserting non-empty ids, and an end-to-end test in `useActiveOutlineItem.test.ts` that runs real extraction output through the spy hook and asserts it doesn't throw.

File: src/hooks/useActiveOutlineItem.ts (hunk 3 above)
Comment (Jayclifford345) [reply src/hooks/useActiveOutlineItem.ts]: Fixed in b605fd0d — reads `root.getBoundingClientRect()` once into `rootRect` and derives both `rootTop` and `bandHeight` from it.

File: src/hooks/useActiveOutlineItem.ts (hunk 4 above)
Comment (Jayclifford345) [reply src/hooks/useActiveOutlineItem.ts:35]: Fixed in b605fd0d — `setActiveId(null)` now runs before the early return when `tracked.length === 0`, matching the `items.length === 0` guard above. Added a regression test covering a content swap to items whose ids don't resolve in the container.

File: src/locales/cs-CZ/grafana-pathfinder-app.json (hunk 1 above)
Comment (Jayclifford345) [reply src/locales/cs-CZ/grafana-pathfinder-app.json]: Fair point — on reflection it's not worth a 20-locale diff for one untranslated aria-label. Removed the `guideReader.outlineLabel` key entirely in cc6ee15a (all locale files + the `t()` call in `OutlineRail.tsx`), hardcoded the aria-label to `"Document outline"` instead. Can revisit with a real i18n pass in a follow-up if this surface grows more user-facing text.

**Review by moxious — APPROVED** [review]
